src/shared/dissect-image.c

```
@@ -150,7 +150,8 @@ static int device_is_partition(sd_device *d, blkid_partition pp) {
                 return false;
 
         r = sd_device_get_sysattr_value(d, "partition", &v);
-        if (r == -ENOENT) /* Not a partition device */
+        if (r == -ENOENT ||        /* Not a partition device */
+            ERRNO_IS_PRIVILEGE(r)) /* Not ready to access? */
```

@bluca Could you check this?

---

Hmm, why would we see EPERM/EACCESS here? the "partition" sysfs attribute is not accessible initially? that's so strange

---

Yeah, but it seems `lstat()` return EPERM.

---

This is so strange... The old code hid that error code I guess. But I wonder how this happens?

---

sorry - no clue, it seems very strange

---

#18717 hides the error again :-p